timesketch/lib/analyzers/contrib/misp_analyzer.py

```diff
@@ -0,0 +1,173 @@
+"""Index analyzer plugin for MISP."""
+
+import logging
+import ntpath
+import requests
+
+from flask import current_app
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+logger = logging.getLogger("timesketch.analyzers.misp")
+
+
+class MispAnalyzer(interface.BaseAnalyzer):
+    """Analyzer for MISP."""
+
+    NAME = "misp_analyzer"
+    DISPLAY_NAME = "MISP"
+    DESCRIPTION = "Mark events using MISP"
+
+    # DEPENDENCIES = frozenset(["domain"])
+
+    def __init__(self, index_name, sketch_id, timeline_id=None, **kwargs):
+        """Initialize the Analyzer.
+
+        Args:
+            index_name: OpenSearch index name
+            sketch_id: The ID of the sketch.
+            timeline_id: The ID of the timeline.
+        """
+        super().__init__(index_name, sketch_id, timeline_id=timeline_id)
+        self.misp_url = kwargs.get("misp_url")
+        self.misp_api_key = kwargs.get("misp_api_key")
+        self.total_event_counter = 0
+        self.request_set = set()
+        self.result_dict = dict()
+
+    @staticmethod
+    def get_kwargs():
+        """Get kwargs for the analyzer.
+
+        Returns:
+            Info to connect to MISP.
+        """
+
+        misp_url = current_app.config.get("MISP_URL")
+        misp_api_key = current_app.config.get("MISP_API_KEY")
+
+        if not misp_api_key or not misp_url:
+            logger.error("MISP conf not found")
+            return []
+
+        matcher_kwargs = [{"misp_url": misp_url, "misp_api_key": misp_api_key}]
+        return matcher_kwargs
+
+    def get_misp_attributes(self, value, attr):
+        """Search event on MISP.
+
+        Args:
+            value:  Can be a valur for: sha1 - sha256 - md5 - filename.
+            attr:  type of the value.
+
+        Returns:
+            List of matching MISP attibutes.
+        """
+        results = requests.post(
+            f"{self.misp_url}/attributes/restSearch/",
+            data={"returnFormat": "json", "value": value, "type": attr},
+            headers={"Authorization": self.misp_api_key},
+            verify=False,
+        )
+
+        if results.status_code != 200:
+            msg_error = "Error with MISP query: Status code"
+            logger.error("{} {}".format(msg_error, results.status_code))
```

Comment:
```suggestion
            logger.error(f"{msg_error} {results.status_code}")
```

Reply:
I have a problem with that, pylint doesn't like this:
`/usr/local/src/timesketch# pylint timesketch/lib/analyzers/contrib/misp_analyzer.py --rcfile .pylintrc -v
************* Module timesketch.lib.analyzers.contrib.misp_analyzer
timesketch/lib/analyzers/contrib/misp_analyzer.py:77:12: W1203: Use lazy % or .format() formatting in logging functions (logging-fstring-interpolation)
`

Reply:
Yes indeed you're right - let's stick to the way it was. Apologies!